server/src/main/java/org/elasticsearch/index/shard/IndexShardOperationPermits.java

```diff
@@ -182,10 +179,14 @@ private void delayOperations() {
     private void releaseDelayedOperations() {
         final List<DelayedOperation> queuedActions;
         synchronized (this) {
```

Comment:
I think we can pull this up to the method. I don't see a reason to drain the queue and the release the queue and it will simplify the reasoning a bit.

Reply:
I'm not sure what you mean. What would you change?

Reply:
I updated my comment. I mean make this entire method synchronized.

Reply:
I find the reasoning simpler here if we don't extend the mutex to a section of the code which it does not need to cover. Are you ok keeping it as is?

Reply:
yes, I'm ok. It's subjective.